### ISSUE TITLE
Erroneous errors when creating generic class instance

### ISSUE THREAD

**Issue by bpatram** [gajus/eslint-plugin-flowtype#343]
Hello, I am using eslint  5.1.0 (current latest) along with eslint-plugin-flowtype 2.50.0 (current latest) using the recommended configuration (`plugin:flowtype/recommended`) and the airbnb preset (`eslint-config-airbnb`)

The code snippet below returns errors which I believe are not correct:

```javascript
// @flow

class MyClass<T> {
  foo: T;

  constructor(foo: T) {
    this.foo = foo;
  }
}

export default new MyClass<string>('test');
```

Running this through the linter returns the following errors:
```
  11:16  error  Missing '()' invoking a constructor  new-parens
  11:27  error  Infix operators must be spaced       space-infix-ops
  11:28  error  'string' is not defined              no-undef
  11:34  error  Infix operators must be spaced       space-infix-ops
```

It seems to be parsing the type def brackets `<` `>` as infix operators GT and LT which causes some issues (such as thinking `string` is a variable and that the constructor never gets invoked).

The above example throws no errors in the [flow type example I made here](https://flow.org/try/#0PTAEAEDMBsHsHcBQiDG0CGBnToCyBPAYQ2wB4AVAPlAG9FRRJZYAuUcgbmQZVgDtMAFwBOAVxSDYwgBRNW7AJS16DUIIAWAS0wA6OaAC8jZlwYBfRBcQBTAB4AHKYNAATa5HSjozvtfh4iEkxSIWFNPgBzSmkAckFrIRiFDiA).

**Comment by gajus** [comment]
There errors are coming from `eslint-config-airbnb`.

**Comment by bpatram** [comment]
@gajus Thanks for responding. Yes, the airbnb preset rules are what's enabling those rules and thus giving me the errors. However they are not correct.

I'm using the babel-eslint parser, so I'm thinking the error is upstream. I will open a ticket with babel-eslint or babel.
